sdk/python/lib/pulumi/resource.py

```diff
@@ -865,9 +875,11 @@ def export(name: str, value: Any):
     :param str name: The name to assign to this output.
     :param Any value: The value of this output.
     """
-    stack = get_root_resource()
-    if stack is not None:
-        stack.output(name, value)
+    res = cast('Stack', get_root_resource())
+    if known_types.is_stack(res):
```

Comment:
does this handle the `None` case from before?

Reply:
Yes. `is_stack` checks for `None`: https://github.com/pulumi/pulumi/blob/9008ac25d834e04a643a1d96b45d6122183af1be/sdk/python/lib/pulumi/runtime/known_types.py#L274